### PR TITLE
[Merged by Bors] - chore: robustifying for debug.byAsSorry (part 2)

### PR DESCRIPTION
**Pull request by kim-em**
Followup to #14993.

Hopefully these are all independently positive or neutral.

These help get Mathlib compiling under `set_option debug.byAsSorry true`.

### CODE REVIEW

File: Mathlib/Order/Minimal.lean [hunk 1]
```diff
@@ -280,6 +282,8 @@ section Image
 variable {f : α → β} {r : α → α → Prop} {s : β → β → Prop}
 
 section
+set_option debug.byAsSorry false
```
Comment (grunweg) [inline Mathlib/Order/Minimal.lean:285]: You want to remove this one, right?

**Review by grunweg — COMMENTED** [review]
These all look reasonable, thanks!
maintainer merge

**Comment by grunweg** [comment]
Um, I didn't check CI yet. As it fails, let me revise this to `maintainer delegate`: once you fixed the build, assuming only simple fixes, feel free to maintainer merge on my behalf.

File: Mathlib/Order/Minimal.lean (hunk 1 above)
Comment (semorrison) [inline Mathlib/Order/Minimal.lean:285]:
```suggestion
```

**Comment by fpvandoorn** [comment]
I'm confused why you changed some term proofs without any tactic proofs. Did you also do some independent cleanup?

This all looks good to me.

bors merge

**Pull request closed without merging**